rts/Sim/MoveTypes/MoveType.cpp

```diff
@@ -129,3 +129,11 @@ bool AMoveType::SetMemberValue(unsigned int memberHash, void* memberValue) {
 	return false;
 }
 
+
+float AMoveType::Waterline() const {
+	// creg
+	if (owner == nullptr)
+		return 0.0f;
+
+	return (owner->unitDef->waterline);
```

Comment:
1) owner can never be `null` at the point where this is called.
2)  initialize `waterline` to `owner->unitDef->waterline` in `AMoveType`'s constructor and return it here.